pyiron_contrib/atomistics/atomicrex/structure_list.py

```diff
@@ -75,24 +91,25 @@ def add_structure(self, structure, identifier):
 
 
     def to_hdf(self, hdf, group_name="flattened_structures"):
-        if self.symbols[-1] == "XX":
-            raise ValueError("Initialized arrays are not filled with values")
+        # truncate arrays to necessary size before writing
+        self._resize_atoms(self.current_atom_index)
```

Comment:
Resizing to current_atom_index truncates the positions of the last structure I think. This should probably be current_atom_index + self.len_current_structure[self.current_structure_index]

Reply:
Nevermind current_atom_index is updated with add_structure so this should work.